tests/unit/api_providers/test_resource_provider.py

```diff
@@ -248,51 +168,53 @@ def test_upload_api_exception_on_version_creation(
         resource_provider.upload("project-id", "name", "type", "path/to/file.txt")
 
 
-def test_upload_failed_file_upload(resource_provider, mock_resources_client):
-    """Test handling of failed upload to presigned URL."""
+@pytest.mark.parametrize(
+    "status_code,error_text,expected_error_msg",
+    [
+        (403, "Forbidden", "Upload failed with status 403: Forbidden"),
+        (500, "Server Error", "Upload failed with status 500: Server Error"),
+    ],
+)
+def test_upload_failed_file_upload(
```

Comment:
Nice!!
Very nitpicky one: Do you reckon we could change the name of this to `test_upload_file_upload_error` just so it matches the naming convention of other tests testing for same errors (ie `test_update_upload_error`) :)
```suggestion
@pytest.mark.parametrize(
    "status_code,error_text,expected_error_msg",
    [
        (403, "Forbidden", "Upload failed with status 403: Forbidden"),
        (500, "Server Error", "Upload failed with status 500: Server Error"),
    ],
)
def test_upload_file_upload_error(
```